src/main/java/io/appform/dropwizard/sharding/dao/operations/lookupdao/GetAndUpdateByLookupKey.java

```diff
@@ -0,0 +1,48 @@
+package io.appform.dropwizard.sharding.dao.operations.lookupdao;
+
+import io.appform.dropwizard.sharding.dao.operations.OpContext;
+import io.appform.dropwizard.sharding.dao.operations.OpType;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+import org.hibernate.Session;
+
+/**
+ * Get an entity from DB by lookup key, mutate it and persist it back to DB. All in same hibernate
+ * session.
+ *
+ * @param <T> Type of entity to get and update.
+ */
+@Data
+@SuperBuilder
+public class GetAndUpdateByLookupKey<T> extends OpContext<Boolean> {
```

Comment:
add test cases for opContext

Reply:
Have added tests for all opcontext classes which has some logic. (ignored the ones which just applies a function)